### ISSUE TITLE
Support custom action for automation rules with webhook + api

### ISSUE THREAD

**Issue by stsewd** [readthedocs/readthedocs.org#6375]
Currently, the actions to be run are only defined in code

https://github.com/readthedocs/readthedocs.org/blob/master/readthedocs/builds/automation_actions.py

Users can't set a custom action. We don't want to execute arbitrary code neither. But we can offer users to set a webhook that will be called on a match. Users can use apiv3 to do whatever they want to the version or project.

### Why don't just use api?

Because users don't have a way to know when the version is created on rtd (nor the final slug). They can only know when the version is created on github.

### Examples

- Match: semver
- action: webhook
- value: https://example.com/webhook

The webhook receive a payload with the project and the matched version. Then the user can trigger a build or deactivate the version or modify other versions, anything.

**Comment by humitos** [comment]
I wouldn't mix the logic for automation rules with outgoing webhooks. Instead, I would add more events to the outgoing webhooks we already have since that should be cleaner UI and we've already created the template replacement there. 

We can add events like `version:created`, `version:deleted`, etc. With that information, the user can perform the action they want on their side based on this data.

<img width="815" height="739" alt="Image" src="https://github.com/user-attachments/assets/600b918f-e481-41c1-86e1-3f45f5ae7ac0" />

**Comment by stsewd** [comment]
I know we also had concerts about users having to set up a server/endpoint to manage the webhook. We could explore to make the webhooks better integrate with webhook dispatch events https://docs.github.com/en/rest/actions/workflows?apiVersion=2026-03-10#create-a-workflow-dispatch-event. I think the only missing thing is being able to pass the Authorization header.

**Comment by humitos** [comment]
> I know we also had concerts about users having to set up a server/endpoint to manage the webhook.

Yes, that's why we built the template system for the webhook body. The use can adapt the body of it as they need.

> We could explore to make the webhooks better integrate with webhook dispatch events [docs.github.com/en/rest/actions/workflows?apiVersion=2026-03-10#create-a-workflow-dispatch-event](https://docs.github.com/en/rest/actions/workflows?apiVersion=2026-03-10#create-a-workflow-dispatch-event).

This would be a completely different feature. We would be changing from "generic webhook" to GitHub specific ones.

**Comment by stsewd** [comment]
> Yes, that's why we built the template system for the webhook body. The use can adapt the body of it as they need.

The concern isn't about the body, it's about having a server to process the events.

> This would be a completely different feature. We would be changing from "generic webhook" to GitHub specific ones.

GitHub actions will be the server that process the webhooks, this isn't about building a direct integration, but providing a way for GH to be able to process the webhooks, in this case, allowing users to define an Authorization header.
